vivisect/impapi/windows/rpcrt4_32.py

```diff
@@ -0,0 +1,522 @@
+    'rpcrt4.char_from_ndr':( 'int', None, 'stdcall', 'rpcrt4.char_from_ndr', (('int', None), ('int', None)) ),
+    'rpcrt4.data_from_ndr':( 'int', None, 'stdcall', 'rpcrt4.data_from_ndr', (('int', None), ('int', None), ('int', None), ('int', None)) ),
+    'rpcrt4.data_into_ndr':( 'int', None, 'stdcall', 'rpcrt4.data_into_ndr', (('int', None), ('int', None), ('int', None), ('int', None)) ),
+    'rpcrt4.data_size_ndr':( 'int', None, 'stdcall', 'rpcrt4.data_size_ndr', (('int', None), ('int', None), ('int', None), ('int', None)) ),
+    'rpcrt4.double_array_from_ndr':( 'int', None, 'stdcall', 'rpcrt4.double_array_from_ndr', (('int', None), ('int', None), ('int', None), ('int', None)) ),
+    'rpcrt4.double_from_ndr':( 'int', None, 'stdcall', 'rpcrt4.double_from_ndr', (('int', None), ('int', None)) ),
+    'rpcrt4.enum_from_ndr':( 'int', None, 'stdcall', 'rpcrt4.enum_from_ndr', (('int', None), ('int', None)) ),
+    'rpcrt4.float_array_from_ndr':( 'int', None, 'stdcall', 'rpcrt4.float_array_from_ndr', (('int', None), ('int', None), ('int', None), ('int', None)) ),
+    'rpcrt4.float_from_ndr':( 'int', None, 'stdcall', 'rpcrt4.float_from_ndr', (('int', None), ('int', None)) ),
+    'rpcrt4.long_array_from_ndr':( 'int', None, 'stdcall', 'rpcrt4.long_array_from_ndr', (('int', None), ('int', None), ('int', None), ('int', None)) ),
+    'rpcrt4.long_from_ndr':( 'int', None, 'stdcall', 'rpcrt4.long_from_ndr', (('int', None), ('int', None)) ),
+    'rpcrt4.long_from_ndr_temp':( 'int', None, 'stdcall', 'rpcrt4.long_from_ndr_temp', (('int', None), ('int', None), ('int', None)) ),
+    'rpcrt4.pfnfreeroutines':( 'int', None, 'cdecl', 'rpcrt4.pfnFreeRoutines', () ),
+    'rpcrt4.pfnmarshallroutines':( 'int', None, 'cdecl', 'rpcrt4.pfnMarshallRoutines', () ),
+    'rpcrt4.pfnsizeroutines':( 'int', None, 'cdecl', 'rpcrt4.pfnSizeRoutines', () ),
+    'rpcrt4.pfnunmarshallroutines':( 'int', None, 'cdecl', 'rpcrt4.pfnUnmarshallRoutines', () ),
+    'rpcrt4.short_array_from_ndr':( 'int', None, 'stdcall', 'rpcrt4.short_array_from_ndr', (('int', None), ('int', None), ('int', None), ('int', None)) ),
+    'rpcrt4.short_from_ndr':( 'int', None, 'stdcall', 'rpcrt4.short_from_ndr', (('int', None), ('int', None)) ),
+    'rpcrt4.short_from_ndr_temp':( 'int', None, 'stdcall', 'rpcrt4.short_from_ndr_temp', (('int', None), ('int', None), ('int', None)) ),
+    'rpcrt4.tree_into_ndr':( 'int', None, 'stdcall', 'rpcrt4.tree_into_ndr', (('int', None), ('int', None), ('int', None), ('int', None)) ),
+    'rpcrt4.tree_peek_ndr':( 'int', None, 'stdcall', 'rpcrt4.tree_peek_ndr', (('int', None), ('int', None), ('int', None), ('int', None)) ),
+    'rpcrt4.tree_size_ndr':( 'int', None, 'stdcall', 'rpcrt4.tree_size_ndr', (('int', None), ('int', None), ('int', None), ('int', None)) ),
+    # FIXME unresolved I_RpcAbortAsyncCall -> RPCRT4.I_RpcAsyncAbortCall
```

Comment:
@upside2 , what do these FIXME's mean?

Reply:
I do not know.  The comments go back 6 years to when our current repository was started.  They were in the original impapi/windows/i386.py file, probably from Visi or Repete.

Reply:
thanks.  again, please accept my apologies for missing the core value of this PR and letting it drop.

this PR currently can't be merged due to an upstream conflict with master.  

the conflict is simple but would you mind pulling in the latest master and fixing the conflict?  
i've done it on my end, but i don't believe i have rights to push to your branch.

Reply:
Yes, I will get to it next week, I hope

Reply:
I submitted a new PR 566 to replace this.  I think that will be easier.

Reply:
for VAMP sigs, yes that was easier.  i just merged #566.

for Import API stuff alone, this PR still has a lot of value.  i really like how you've broken out the many areas of impapi sections, and support more msvc* libraries.
also, your changes before were cleaner and prettier than what we had for MSVS Vamp.

i really hated adding in whether a function returned void, but after reviewing it again i understand why.  my mindset when you first submitted it was in heavy full-emulation, so it didn't click with me.  at that point i was wildly overwhelmed, and i recoiled (i apologize).  looking at how you used it in impapi for emulation and symboliks, i get it now.

if you are interested, this PR represents a lot of work, and i'd like to merge it in and have you get credit.  i may create a PR separately if you don't want to, and @rakuy0 will have to review it.

but if you want the win, it doesn't seem like a lot of work to even create a new branch and copy over the work from this one, if the merge-collisions are too annoying.

either way, thank you for your contribution.

Reply:
Thanks for the kind words!  I will separate the two logical sections as new branches; this original PR had too many things thrown into it.  I will also be sending a couple other fixes soon.